### ISSUE TITLE
The wrong subtitle gets matched in archives

### ISSUE THREAD

**Issue by ThisIsntTheWay** [morpheus65535/bazarr#2578]
**Describe the bug**
When trying to match subtitles from an archive, [`_get_matching_sub()`](https://github.com/morpheus65535/bazarr/blob/b4195934c1fd5ff97acf07d7fbd82f52e272c520/custom_libs/subliminal_patch/providers/utils.py#L45) won't match the correct one.
```python
sub_name = "ONE PIECE S01E31 sdh-jpn.srt"
guess_options = {'single_value': True, 'type': 'episode'}
guess = guessit(sub_name, options=guess_options)
# MatchesDict([('title', 'ONE PIECE'), ('season', 1), ('episode', 31), ('episode_title', 'sdh'), ('subtitle_language', <Language [ja]>), ('container', 'srt'), ('mimetype', 'text/plain'), ('type', 'episode')])
```

I invoke the function using:
```python
get_subtitle_from_archive(
   archive, 
   episode=subtitle.video.episode,
   episode_title=subtitle.video.title
)
```

Result:
`Matches: [_MatchingSub(file='ONE PIECE S01E01 sdh-jpn.srt', priority=2, context='Episode number matched')]`

Also I guess this should be `if not ...` as it will write to log when an episode _was_ found:  
https://github.com/morpheus65535/bazarr/blob/b4195934c1fd5ff97acf07d7fbd82f52e272c520/custom_libs/subliminal_patch/providers/utils.py#L67-L69

**To Reproduce**
1. List subtitles for an episode
2. Download an archive
3. Bazarr will choose the wrong sub for the given episode, usually the first file in the archive.

**Expected behavior**
Bazarr chooses the subtitle whose episode number matches with whatever `episode` in `_get_matching_sub()` is set to.

**Software (please complete the following information):**
 - Bazarr: v1.4.4-beta.23 / Commit b419593

**Comment by vitiko98** [comment]
Please give more context

> 
> ```python
> get_subtitle_from_archive(
>    archive, 
>    episode=subtitle.video.episode,
>    episode_title=subtitle.video.title
> )
> ```

What is the content of `subtitle`? What is the archive?

**Comment by ThisIsntTheWay** [comment]
`archive` is an object as returned by `get_archive_from_bytes()`, therefore a `RarFile` or `ZipFile`.  
`subtitle` is a provider specific `Subtitle` object, but the property being used is really just a `video`.

The relevant code can be [found here](https://github.com/ThisIsntTheWay/bazarr/blob/5db53f674b37b2134e9a62e08546d2def2471a00/custom_libs/subliminal_patch/providers/jimaku.py#L263-L270).

**Comment by vitiko98** [comment]
I'm talking about the actual content of the objects. You can provide debug logs or point me the subtitle link and the episode information.

**Comment by ThisIsntTheWay** [comment]
Turns out this has been due to user error...  

My provider script fetches subs from a website that potentially uses absolute episode numbers instead of the usual season/episode format.
I added some logic to translate season/episode to absolute episode numbers, but when executing `get_subtitle_from_archive()` I fail to pass the adjusted episode number.
So, for a show whose `S03E01` would translate into episode `31`, the matcher uses episode `1` instead of `31`.

I guess the only useful takeaway from this issue is that `if matched_episode_num:` should be negated.
